command/pki_health_check.go

```diff
@@ -223,10 +223,22 @@ func (c *PKIHealthCheckCommand) Run(args []string) int {
 
 	// Handle listing, if necessary.
 	if c.flagList {
-		c.UI.Output("Default health check config:")
+		uiFormat := Format(c.UI)
+		if uiFormat == "yaml" {
+			c.UI.Error("YAML output format is not supported by the --list command")
+			return pkiRetUsage
+		}
+
+		if uiFormat != "json" {
+			c.UI.Output("Default health check config:")
+		}
 		config := map[string]map[string]interface{}{}
 		for _, checker := range executor.Checkers {
 			config[checker.Name()] = checker.DefaultConfig()
+			// When generating the configuration, add the defaults for the enabled flag
```

Comment:
Should we do this in Executor?

Reply:
I've been convinced this was a bad idea as the configuration generated by this will override the `-default-disabled` flag which has a valuable use case say in CI pipelines where end-users could specify individual tests they would want to run.

Will remove this.